pennylane/operation.py

```diff
@@ -798,6 +850,53 @@ class Observable(Operator):
     # pylint: disable=abstract-method
     return_type = None
 
+    @classmethod
+    def _eigvals(cls, *params):
+        """Eigenvalues of the observable.
+
+        The order of the eigenvalues needs to match the order of
+        the computational basis vectors when the observable is
+        diagonalized using :attr:`diagonalizing_gates`.
+
+        This is a *class method* that must be defined for all
+        new diagonal operations, that returns the eigenvalues
+        of the operator in the computational basis.
+
+        This private method allows eigenvalues to be computed
+        directly without instantiating the operators first.
+
+        To return the eigenvalues of *instantiated* operators,
+        please use the :attr:`~.Operator.eigvals` property instead.
+
+        **Example:**
+
+        >>> qml.PauliZ._eigvals()
+        >>> array([1, -1])
+
+        Returns:
+            array: eigenvalue representation
+        """
+        raise NotImplementedError
+
+    @property
+    def eigvals(self):
+        r"""Eigenvalues of an instantiated observable.
+
+        The order of the eigenvalues needs to match the order of
+        the computational basis vectors when the observable is
+        diagonalized using :attr:`diagonalizing_gates`.
```

Comment:
(Minor suggestion): great description! just thought of adding this point to refer to two things:
* this does not hold for `CV`
* an observable can be defined with an arbitrary order of eigenvalues (and potentially without diagonalizing gates), but then it cannot be used in a quantum function
```suggestion
        diagonalized using :attr:`diagonalizing_gates`. This is a requirement for using qubit observables in quantum functions.
```